src/apps/Hosting/Files.js

```diff
@@ -121,19 +202,63 @@ const HostingFilesView = React.createClass({
     return file;
   },
 
+  renderNewFolderButtons() {
+    const { name, errors, showNewFolderButton } = this.state;
+    const hasErrors = errors.name && errors.name.length;
+    const styles = this.getStyles();
+    const textFieldStyle = hasErrors ? { marginBottom: 22 } : null;
+
+    if (showNewFolderButton && !hasErrors) {
+      return (
+        <RaisedButton
+          label="Create new folder"
+          primary={true}
+          style={styles.newFolderButton}
+          onTouchTap={this.handleClickNewFolderButton}
+        />
+      );
+    }
+
+    return (
+      <div style={styles.newFolderForm}>
+        <TextField
+          fullWidth={true}
+          name="name"
+          value={name}
+          onChange={this.handleNewFolderNameChange}
+          errorText={this.getValidationMessages('name').join(' ')}
+          hintText="Type new folder name"
+          style={{ ...styles.newFolderNameInput, ...textFieldStyle }}
+        />
+        <RaisedButton
```

Comment:
You can use one component for both buttons and control it via props. I mean buttons for `Create new folder` and `Create`